src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3046,6 +3046,15 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
                 {
                     var existingOrAddedMembers = addProperties(ctor.Parameters);
                     addDeconstruct(ctor, existingOrAddedMembers);
+
+                    foreach (var parameter in ctor.Parameters)
```

Comment:
>foreach (var parameter in ctor.Parameters) [](start = 20, length = 42)

I am not sure this is the right place for the check and the right set of members to check. If I remember correctly, any field in the type is a subject for equality.

Reply:
Good catch. Thanks

---
In reply to: [512787196](https://github.com/dotnet/roslyn/pull/48925#discussion_r512787196) [](ancestors = 512787196)